bentoml/_internal/frameworks/catboost.py

```
@@ -0,0 +1,273 @@
+
+    with bentoml.models.create(
+        name,
+        module=MODULE_NAME,
+        api_version=API_VERSION,
+        signatures=signatures,
+        labels=labels,
+        custom_objects=custom_objects,
+        metadata=metadata,
+        context=context,
+    ) as bento_model:
+        model.save_model(bento_model.path_of(MODEL_FILENAME))  # type: ignore (incomplete CatBoost types)
+
+        return bento_model
+
+
+def get_runnable(bento_model: bentoml.Model) -> t.Type[bentoml.Runnable]:
+    """
+    Private API: use :obj:`~bentoml.Model.to_runnable` instead.
+    """
+
+    class CatBoostRunnable(bentoml.Runnable):
+        SUPPORT_NVIDIA_GPU = True
+        SUPPORT_CPU_MULTI_THREADING = True
+
+        predict_params: t.Dict[str, t.Any]
+
+        def __init__(self):
+            super().__init__()
+            self.model = load_model(bento_model)
+
+            self.predict_params = {
+                "task_type": "CPU",
+            }
+
+            # check for resources
+            available_gpus = os.getenv("NVIDIA_VISIBLE_DEVICES")
+            if available_gpus is not None and available_gpus != "":
+                self.predict_params["task_type"] = "GPU"
+            else:
+                nthreads = os.getenv("OMP_NUM_THREADS")
+                if nthreads is not None and nthreads != "":
+                    nthreads = max(int(nthreads), 1)
+                else:
+                    nthreads = -1
+                self.predict_params["thread_count"] = nthreads
+
+            self.predict_fns: dict[str, t.Callable[..., t.Any]] = {}
+            for method_name in bento_model.info.signatures:
+                try:
+                    self.predict_fns[method_name] = getattr(self.model, method_name)
+                except AttributeError:
+                    raise InvalidArgument(
+                        f"No method with name {method_name} found for CatBoost model of type {self.model.__class__}"
+                    )
```

Does this mean that a CatBoost model can have multiple method_name?

---

Can we have a test for all possible method_names? And if if the provided method_name is not in possible method_names, we should raise an exception?

---

@larme I have added all the method names to the test. As well as this code I copied from xgboost, which throws an exception

---

Something we should circle back around and add to XGBoost for sure.